lib/csv.ts

```diff
@@ -0,0 +1,136 @@
+export type CsvContent = {
+  filename: string
+  headerRow: string
+  body: string
+}
+export const escapeCSVValue = (value: string) => {
+  if (
+    value.includes(',') ||
+    value.includes('"') ||
+    value.includes('\r') ||
+    value.includes('\n')
+  ) {
+    return `"${value.replaceAll('"', '""')}"`
+  }
+  return value
+}
+export function createCSV({
+  headers,
+  lines,
+  filename,
+  lineDelimiter = '\n',
+}: {
+  lines: string[]
+  headers: string[]
+  filename?: string
+  lineDelimiter?: string
+}) {
+  return {
+    filename: (filename || Date.now().toString()) + '.csv',
+    headerRow: headers.join(',') + lineDelimiter, // make your own csv head
+    body: lines.join(lineDelimiter),
+  }
+}
```

Comment:
The return value does not match the returned type. Is there a particular reason for that ?

The data structure to represent the CSV here is a mix between encoded data (a multi line string of comma separated values, with the firt line being a header) and "parsed" data (internal data structure), Looks like we could benefit from defining and using a type interface (or even a class ?) for internal representation of CSVs:

```ts
type CsvContent = {
  headers: string[],
  rows: [][]
}
```

Reply:
> The return value does not match the returned type

not sure if I follow? the return type is expected to be `CsvContent` which i left implicit I guess. agreed we can tidy this up a bit more to make a more strict definition but for now the use-case is super low-stake so wanted to keep this flexible/open.